Skip ALL granularity compaction (#13304)
apache/druid: docs/data-management/automatic-compaction.md

}
```

> Please note that auto-compaction currently skips ALL granularity intervals when the target segment granularity is finer.

Reviewer: Maybe put this as a quoted section:
e.g.
```suggestion
> Auto-compaction skips segments with granularity ALL if the target segment granularity is not ALL.
```
